### PR TITLE
Add similar documents query to index 

### LINKED ISSUE #755: [v1.9] Get similar documents
*Following this [central issue](https://github.com/meilisearch/integration-guides/issues/301)*

This is related to a newly introduced feature in [Meilisearch v1.9.0](https://github.com/meilisearch/meilisearch/releases/tag/v1.9.0): get similar documents.

Refer to [docs for more information](https://www.meilisearch.com/docs/reference/api/similar#get-similar-documents-with-post).

- [ ] Add a new method named `searchSimilarDocuments()` (adapt the case to the language) having 1 parameter. This parameter is an object corresponding to the body you should pass to the `POST /similar` route. Do not manage error, Meilisearch will handle them if a field is missing in the body.
Do NOT implement the route for `GET /indexes/{index_uid}/similar`, we will add it in a second iteration if requested.
- [ ] Add tests. Here is the [basic test](https://github.com/meilisearch/meilisearch-php/pull/645#discussion_r1632879832) to do to check everything is fine
- [ ] Update the `.code-samples.meilisearch.yaml`
    - [ ] Add `get_similar_post_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/04f26a35771305378ed0a506f4d60ef06d07bd7e/.code-samples.meilisearch.yaml#L1333-L1340

### PR DESCRIPTION
**Pull request by jdvalenzuelah**
# Pull Request

## Related issue
Fixes #755

## What does this PR do?
- Add similar documents method to index

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Review by curquiza — CHANGES_REQUESTED** [review]
Hello @jdvalenzuelah 

can you fix your test `testSimilarDocuments`? The CI is red right now 😊

**Comment by jdvalenzuelah** [comment]
Hi @curquiza is there a way to get a full stacktrace in this run? Test are passing for me locally and don't really understand where the exception is coming from.

**Comment by curquiza** [comment]
@jdvalenzuelah 

Looks like there is an issue in the test you created: 

```
 SearchTest > testSimilarDocuments() FAILED
    java.lang.ClassCastException at SearchTest.java:849
```

**Comment by jdvalenzuelah** [comment]
Hi @curquiza I think test should work in CI now. 

When running locally i enable `vectorStore` using curl but that will obviously not be enabled in the image used in CI.

**Comment by curquiza** [comment]
Hello @jdvalenzuelah 
The tests still don't pass, including the linter.
You can click on them to have more information

Capture d’écran 2024-08-15 à 14 11 44

**Comment by jdvalenzuelah** [comment]
> Hello @jdvalenzuelah The tests still don't pass, including the linter. You can click on them to have more information
> Capture d’écran 2024-08-15 à 14 11 44

Fixed the linter errors. The tests did not run because the `spotlessCheck` task failed.

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -822,3 +822,8 @@ multi_search_1: |-
   multiIndexSearch.addQuery(new IndexSearchRequest("movie_ratings").setQuery("us"));
 
   client.multiSearch(multiSearchRequest);
+get_similar_post_1:
+  SimilarDocumentRequest query =  new SimilarDocumentRequest()
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:826]:
```suggestion
  SimilarDocumentRequest query = new SimilarDocumentRequest()
```

**Review by curquiza — CHANGES_REQUESTED** [review]
Nice

**Review by curquiza — APPROVED** [review]
bors merge

**Pull request merged**